test/e2e/specs/editor/blocks/links.spec.js

```diff
@@ -8,6 +8,537 @@ test.describe( 'Links', () => {
 		await admin.createNewPost();
 	} );
 
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test.use( {
+		LinkUtils: async ( { editor, page, pageUtils }, use ) => {
+			await use( new LinkUtils( { editor, page, pageUtils } ) );
+		},
+	} );
+
+	test( `will use Post title as link text if link to existing post is created without any text selected`, async ( {
+		admin,
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		const titleText = 'Post to create a link to';
+		await admin.createNewPost( { title: titleText } );
+		const postId = await editor.publishPost();
+		await admin.createNewPost();
+
+		// Now in a new post and try to create a link from an autocomplete suggestion using the keyboard.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type( 'Here comes a link: ' );
+
+		// Press Cmd+K to insert a link.
+		await pageUtils.pressKeys( 'primary+K' );
+
+		// Trigger the autocomplete suggestion list and select the first suggestion.
+		await page.keyboard.type( titleText.substr( 0, titleText.length - 2 ) );
```

Comment:
Nit: Whilst this is valid it does feel very readable. In general I would lean towards prefer concretions over abstractions in tests.
```suggestion
		await page.keyboard.type( 'Post to create a' );
```

Reply:
✅